src/components/form/form.js

```diff
@@ -518,14 +532,21 @@ class Form extends React.Component {
     if (!valid) { ev.preventDefault(); }
 
     if (this.props.afterFormValidation) {
-      this.props.afterFormValidation(ev, valid);
+      this.props.afterFormValidation(ev, valid, this.enableForm);
     }
 
     if (valid && this.props.onSubmit) {
-      this.props.onSubmit(ev);
+      this.props.onSubmit(ev, this.enableForm);
+      if (this.props.autoDisable) {
+        this.setState({ submitted: true });
```

Comment:
prefer this where you had it as `this.validate` might be slow if it is a big form - the sooner we disable it the better.

Reply:
I don't understand what you mean here. You mean put it further up ? If so, surely we have to make sure that the form has validated before disabling the save button. If not the use would try and send with invalid info, then when he corrects the info and tries to resend the button will be disabled. Or have I not understood your demand